pkg/skaffold/schema/defaults/defaults_test.go

```diff
@@ -254,7 +254,7 @@ func TestSetDefaultPortForwardNamespace(t *testing.T) {
 	err := Set(cfg)
 	testutil.CheckError(t, false, err)
 	testutil.CheckDeepEqual(t, "mynamespace", cfg.PortForward[0].Namespace)
-	testutil.CheckDeepEqual(t, constants.DefaultPortForwardNamespace, cfg.PortForward[1].Namespace)
+	testutil.CheckDeepEqual(t, "ns", cfg.PortForward[1].Namespace)
```

Comment:
umm, shdn't this be 
1. default namepsace when no currentNamespace is specified.
2. whatever is the current namespace when currentnamespace is specified.

Can you  override `currentNamespace` and mock it in test `**TestSetDefaultPortForwardNamespace**`?
Add 2 more test cases covering above 2 scenarios.

Reply:
The SetupFakeKubernetesContext function has a behaviour I do not understand, as It seems not possible to change the current context by executing it a second time with a different context. Do I miss something?

Reply:
```
func getRawKubeConfig() (clientcmdapi.Config, error) {
	kubeConfigOnce.Do(func() {
```

In fact, it is not possible to update the kubeconfig file during the tests execution to override the currentContext, because the kubeconfig file is cached.

@tejal29 can you advise me how to handle this?

Reply:
@tejal29 thanks, I've added the tests